platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/assertions/MultipleNodeReconnectResultContinuousAssert.java

```diff
@@ -0,0 +1,148 @@
+     *
+     * @param actual the actual {@link MultipleNodeReconnectResults} to assert
+     */
+    public MultipleNodeReconnectResultContinuousAssert(@Nullable final MultipleNodeReconnectResults actual) {
+        super(actual, MultipleNodeReconnectResultContinuousAssert.class);
+    }
+
+    /**
+     * Creates a continuous assertion for the given {@link MultipleNodeReconnectResults}.
+     *
+     * @param actual the {@link MultipleNodeReconnectResults} to assert
+     * @return a continuous assertion for the given {@link MultipleNodeReconnectResults}
+     */
+    @NonNull
+    public static MultipleNodeReconnectResultContinuousAssert assertContinuouslyThat(
+            @Nullable final MultipleNodeReconnectResults actual) {
+        return new MultipleNodeReconnectResultContinuousAssert(actual);
+    }
+
+    /**
+     * Asserts that the node has no failed reconnects.
+     *
+     * @return a continuous assertion for the given {@link SingleNodeReconnectResult}
+     */
+    @NonNull
+    public MultipleNodeReconnectResultContinuousAssert hasNoFailedReconnects() {
+        return checkContinuously(logEntry -> {
+            if (logEntry.message().contains(ReconnectFailurePayload.class.toString())) {
+                failWithMessage(
+                        "Expected no failed reconnects, but found %s on node %s",
+                        logEntry.message(), logEntry.nodeId());
+            }
+        });
+    }
+
+    /**
+     * Asserts that the node does attempt to perform any reconnects.
+     *
+     * @return a continuous assertion for the given {@link SingleNodeReconnectResult}
+     */
+    @NonNull
+    public MultipleNodeReconnectResultContinuousAssert doNotAttemptToReconnect() {
+        return checkContinuously((logEntry) -> {
+            if (logEntry.message().contains(ReconnectStartPayload.class.toString())) {
+                failWithMessage(
+                        "Expected no failed reconnect, but found %s on node %s", logEntry.message(), logEntry.nodeId());
+            }
+        });
+    }
+
+    /**
+     * Asserts that the nodes have no reconnects that take longer than the provided time.
+     *
+     * @param maximumReconnectTime the maximum allowed reconnect time
+     * @return a continuous assertion for the given {@link MultipleNodeReconnectResultContinuousAssert}
+     */
+    @NonNull
+    public MultipleNodeReconnectResultContinuousAssert haveMaximumReconnectTime(
+            @NonNull final Duration maximumReconnectTime) {
+        isNotNull();
+        return checkContinuously(logEntry -> {
+            if (logEntry.message().contains(SynchronizationCompletePayload.class.toString())) {
+                final SynchronizationCompletePayload payload =
+                        parsePayload(SynchronizationCompletePayload.class, logEntry.message());
+                if (payload.getTimeInSeconds() > maximumReconnectTime.getSeconds()) {
```

Comment:
It's unfortunate that `payload.getTimeInSeconds()` returns a `double`. We should either implement an exact comparison or make it clear that we are rounding to full seconds. The same goes for `haveMaximumTreeInitializationTime()`.